source/scripts/Background/controllers/MessageHandler/requests.ts

```diff
@@ -85,10 +79,18 @@ export const handleRequest = async (
                 (ev: any) => {
                     console.log('Connect window addEventListener', ev.detail);
                     if (ev.detail.windowId === windowId) {
-                        port.postMessage({
-                            id: message.id,
-                            data: {result: ev.detail.result, data: ev.detail.signature},
-                        });
+                        
+                        if (ev.detail.result) {
+                            port.postMessage({
+                                id: message.id,
+                                data: ev.detail.signature.hex,
+                            });
+                        } else {
```

Comment:
Didn't we need the public key for DAG signing?

Reply:
Yes, we need it... We get it from dag_getPublicKey...
Nevertheless, after digging more on the topic it will be possible to get the public key from the message and signature (after I study a little bit further the math of ECDSA, I just need a little bit more time, but eventually, for sure).